### PR TITLE
gh-128192: mark new tests with skips based on hashlib algorithm availability

### PR DESCRIPTION
**Pull request by gpshead**
This is not _important_. The buildbot this is ameliorating is a crazy configuration that we don't support, primarily for testing purposes. This just tries to maintain pairity with which oddball bots were failing before the issues main change rather than regressing one.

* Issue: gh-128192

### CODE REVIEW

**Comment by gpshead** [comment]
!buildbot FIPS

**Comment by gpshead** [comment]
!buildbot FIPS

**Comment by gpshead** [comment]
!buildbot Blake2

File: Lib/test/test_urllib2.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import urllib.error
 import http.client
 
+
```
Comment (picnixz) [inline Lib/test/test_urllib2.py:30]: I'm not sure if this was desired to be part of the diff.

**Review by picnixz — APPROVED** [review]
LGTM. Thanks for doing it!

**Pull request merged**